train.py

```diff
@@ -183,7 +183,8 @@ def __init__(self, hparams_, datapath, batch_size, warmup_steps=4000,
         self.val_dl = DataLoader(dataset=self.val_ds, batch_size=batch_size, shuffle=True)
 
         # create model
-        self.model = MusicTransformer(**hparams_).to(device)
+        model = MusicTransformer(**hparams_).to(device) 
+        self.model = torch.compile(model)
```

Comment:
This prevents the model from being saved and loaded, the way the trainer is currently written, I think it should remain as it was before, and the compile step should occur in the `fit` method. That way, it would have an extra compilation overhead when it first starts, or loads from a checkpoint, but we would still be able to use the Trainer properly. See: https://discuss.pytorch.org/t/how-to-save-load-a-model-with-torch-compile/179739 - currently compiled models cannot be properly saved or loaded.
```suggestion
        self.model = MusicTransformer(**hparams_).to(device) 
```

Reply:
Will do

Reply:
Fixed.